addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/handler/HarmonyHubHandler.java

```diff
@@ -233,27 +205,57 @@ private synchronized void connect() {
             }
         }
 
-        HarmonyClient localClient = HarmonyClient.getInstance();
-        localClient.addListener(this);
-        client = localClient;
+        HarmonyClient localClient = new HarmonyClient();
+        harmonyClientListener = new HarmonyClientListener() {
+
+            @Override
+            public void hubDisconnected(@Nullable String reason) {
+                setOfflineAndReconnect(String.format("Could not connect: %s", reason));
+
+            }
+
+            @Override
+            public void hubConnected() {
+                client = localClient;
+                heartBeatJob = scheduler.scheduleWithFixedDelay(() -> {
+                    try {
+                        client.sendPing();
+                    } catch (Exception e) {
+                        logger.warn("heartbeat failed", e);
```

Comment:
It's better to change log the stacktrace only on debug since the Thing is already set offline with a reason on the line below.